libnet/src/ioctl.rs

```diff
@@ -26,25 +26,52 @@ use std::os::unix::io::AsRawFd;
 use std::ptr;
 use tracing::{debug, warn};
 
+// A wrapper to ensure that the argument passed to ioctl is mutable
+unsafe fn wrap_ioctl_rw<T>(fd: c_int, req: c_int, data: *mut T) -> c_int {
```

Comment:
Something that might simplify some of the code:

`as_raw_fd` on Unix platforms is provided by the `std::os::fd::AsRawFd` trait. This is implemented for, among other types, `c_int`.

Meaning, this routine could take its first parameter as

```rust
fd: impl AsRawFd
```

and apply to both `c_int` and, say, `&Socket`. This might let you collapse the `ioctl!`/`rioctl!` distinction.

Reply:
Good call, I've collapsed `ioctl!`/`rioctl!` in this manner.